client/src/components/my-mentor-card/use-with-recommended-action.tsx

```diff
@@ -189,22 +215,32 @@ export function UseWithRecommendedAction(
     mentor?.answers.filter((a) => a.status === Status.COMPLETE).length || 0;
   const totalAnswers = mentor?.answers.length || 0;
 
+  return {
+    mentorId: mentor?._id,
```

Comment:
same issue about narrowing mentor. It would be better to have a check at the top of this function

```typescript
if(!mentor) { // compile-tie types say this is impossible, but if you want to handle it for runtime
  return { ... } // whatever makes sense for the (unexpected) no-mentor case
}